### PR TITLE
Switch multi-optimizer tests to manual optimization

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Splits out test changes from #16539.
#16539 will simplify the automatic optimization loop, dropping support for multiple optimizers. Instead, users who need to handle multiple optimizers will switch to manual optimization for greater flexibility.
This PR converts a subset of the tests that run with multiple optimizers to make use of manual optimization instead.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: tests/tests_pytorch/callbacks/test_lr_monitor.py [hunk 1]
```diff
@@ -263,8 +282,22 @@ def test_lr_monitor_no_lr_scheduler_multi_lrs(tmpdir, logging_interval: str):
...
+            super().__init__()
+            self.automatic_optimization = False
+
+        def training_step(self, batch, batch_idx):
+            opt1, opt2 = self.optimizers()
+
+            loss = self.loss(self.step(batch))
+            self.manual_backward(loss)
+            opt1.step()
+            opt1.zero_grad()
```
Comment (carmocca) [inline tests/tests_pytorch/callbacks/test_lr_monitor.py:295]: nit (applies to all changes): in automatic optimization we do zero_grad first

File: tests/tests_pytorch/trainer/optimization/test_optimizers.py [hunk 2]
```diff
@@ -735,15 +729,10 @@ def configure_optimizers(self):
...
-
-    with patch.object(CustomEpochScheduler, "step") as mock_method_epoch, patch.object(
-        torch.optim.lr_scheduler.StepLR, "step"
-    ) as mock_method_step:
+    with mock.patch.object(CustomEpochScheduler, "step") as mock_method_epoch:
         trainer.fit(model)
 
     assert mock_method_epoch.mock_calls == [call(epoch=e) for e in range(max_epochs)]
-    # first step is called by PyTorch LRScheduler
-    assert mock_method_step.call_count == max_epochs * limit_train_batches + 1
```
Comment (carmocca) [inline tests/tests_pytorch/trainer/optimization/test_optimizers.py:746]: can we keep the step-level scheduler check? Maybe parametrizing this test or writing a duplicate that sets the scheduler on a step interval

**Review by carmocca — APPROVED** [review]

File: tests/tests_pytorch/trainer/optimization/test_optimizers.py (hunk 2 above)
Comment (awaelchli) [reply tests/tests_pytorch/trainer/optimization/test_optimizers.py]: Actually, the simplest seems to be to just keep all assertions but reduce the number of optimizers to 1.
